src/compiler/checker.ts

```diff
@@ -37577,14 +37577,14 @@ namespace ts {
                         break;
 
                     case SyntaxKind.PropertyDeclaration:
-                        if (!hasModifier(parent, ModifierFlags.Static) ||
-                            !hasModifier(parent, ModifierFlags.Readonly)) {
+                        if (!hasSyntacticModifier(parent, ModifierFlags.Static) ||
+                            !hasSyntacticModifier(parent, ModifierFlags.Readonly)) {

```

Comment:
shouldn't this check for jsdoc `@readonly`?

Reply:
This is a grammar check for `unique symbol`, which doesn't seem to be executed for this case anyways:
```ts
// @ts-check
class C {
    /** 
     * @type {unique symbol} // ERROR: 'unique symbol' types are not allowed here.
     */
    static x;
}
```

It doesn't work currently because `checkGrammarTypeOperatorNode` only walks up parenthesized types on the parent, but doesn't walk up JSDoc types currently. We might want to fix this but it may be in a separate issue/PR?

Reply:
The fix is small enough I can add it to this as well.